packages/react/src/primitives/Link/Link.tsx

```diff
@@ -4,11 +4,16 @@ import classNames from 'classnames';
 import { ComponentClassNames } from '../shared';
 import { LinkProps, Primitive } from '../types';
 import { View } from '../View';
+import { useDeprecationWarning } from '../../hooks/useDeprecationWarning';
 
 const LinkPrimitive: Primitive<LinkProps, 'a'> = (
   { as = 'a', children, className, isExternal, to, ...rest },
   ref
 ) => {
+  useDeprecationWarning({
+    shouldWarn: to != null,
+    message: `Please stop using 'to' prop as it will be deprecated soon.`,
```

Comment:
Should we suggest an alternative here?

Reply:
Yeah, I can suggest people to use our `Link` as a child component of the Link from other routing libs, and use `to` prop on that Link instead